### ISSUE TITLE
babel-runtime in 0.27.1

### ISSUE THREAD

**Issue by JiLiZART** [acdlite/recompose#714]
Hi, I noticed that in `0.27.1` was introduced `babel-runtime` dep.
![image](https://user-images.githubusercontent.com/62051/43388036-5ea9b6ee-93e8-11e8-89e7-947a874bcb8e.png)

Is there a really critical package for 0.27.x? or it can be solved by using more relative and small package?

And it cost me + 12kb in my bundle.js

https://packagephobia.now.sh/result?p=recompose@0.26.0
![image](https://user-images.githubusercontent.com/62051/43388693-1b939f44-93ea-11e8-825f-8cab0bd9a0f6.png)

vs

https://packagephobia.now.sh/result?p=recompose@0.27.1
![image](https://user-images.githubusercontent.com/62051/43388725-2fb367b6-93ea-11e8-9e43-3e338bd630af.png)

now I downgraded to 0.26.0

**Comment by istarkov** [comment]
cc @TrySound

**Comment by TrySound** [comment]
@JiLiZART babel runtime is not the problem. With the right configuration it's a good option to reuse helpers across packages. I just upgraded to v7 and disabled pollyfills. The size is significantly reduced.
https://github.com/acdlite/recompose/pull/715/files#diff-29fd455fe4a1c8d19c5e5b1420ce14faR1

**Comment by TrySound** [comment]
@JiLiZART Do you have +12kb in production webpack v4 build?

**Comment by JiLiZART** [comment]
@TrySound I don't use babel-runtime at all. I use a few polyfills from `core-js`. 
Production optimized + gzip bundle with babel-runtime cost me + 3kb (its a huge for the site and its business conversion).
`babel-runtime` includes whole `core-js` library, its bad for me :(

**Comment by TrySound** [comment]
@JiLiZART Not anymore. And not before. Previously necessary stuff was cherry picked. For the next release I disabled polyfills from core-js at all. There will be only deduplicated helpers.

**Comment by TrySound** [comment]
@JiLiZART By the way do you use babel?

**Comment by TrySound** [comment]
Ah, sorry. Misread. babel-runtime is useful when it's reused across your dependencies. It's not necessary to have babel in your pipeline.

**Comment by JiLiZART** [comment]
Yep, when it ships as peer dependency. I think its a good way.
All the packages that I use with `babel-runtime` as dep. don't ship it into production bundle. Because of size matters.

**Comment by TrySound** [comment]
If they don't use babel-runtime then they inline all helpers so they are duplicated in every package. babel-runtime solves this. Without polyfills it's the same as what bundle usually do. So there is a chance that some helpers will be reused.

**Comment by TrySound** [comment]
That's the point of babel runtime. Size matters.

**Comment by TrySound** [comment]
Peer dependency can lead to broken packages because they may use different versions of babel runtime.

**Comment by istarkov** [comment]
Thanks 2 @TrySound new 0.28.0 has released

**Comment by TrySound** [comment]
Here's what bundle phobia shows me

![image](https://user-images.githubusercontent.com/5635476/43397130-705afd0c-940c-11e8-994b-f7de2677f322.png)

**Comment by JiLiZART** [comment]
As they use different versions of babel runtime, it can bundled with both versions of the same package.
Thanks for the help! +0.6kb cost acceptable for me!

**Comment by TrySound** [comment]
Yes, safely use both versions as if helpers was bundled in every packages. But still there will be packages with the same version which could reuse it.

**Comment by TrySound** [comment]
This cost is related to new recompose code and new babel helpers (transpiling artefacts, not polyfills).
